shopBuddy/shopBuddy/BarcodeAPIManager.m

```diff
@@ -56,15 +55,39 @@ - (void)searchItemsWithQuery:(NSString *)search completion:(void(^)(NSMutableArr
 {
 
     NSString *const path = [NSString stringWithFormat:kSearch_url, search, key];
+    __weak __typeof__(self) weakSelf = self;
+    __block NSMutableArray *all_items = [NSMutableArray new];
+    [manager GET:path parameters:nil headers: nil progress:nil success:^(NSURLSessionTask *task, NSDictionary *responseObject)
+     {
+        for (int i = 0; i < [responseObject[@"products"] count]; i++)
+        {
+            Item *item = [Item createItemWithDictionary:responseObject[@"products"][i]];
+            NSArray *categories = [responseObject[@"products"][i][@"category"] componentsSeparatedByString:@" > "];
+            item.category = categories[0];
+            [all_items addObject:item];
+            
+        }
+        [weakSelf _getNextPageWithCursor:responseObject[@"metadata"] [@"next_cursor"] withQuery:search completion:^(NSMutableArray<Item *> *items, NSError *error) {
+            [all_items addObjectsFromArray:items];
+            completion(all_items, nil);
+        }];
+     }failure:^(NSURLSessionDataTask *task, NSError *error)
+     {
+        completion(nil,error);
+     }];
+}
 
+- (void)_getNextPageWithCursor:(NSString *)cursor withQuery:(NSString *)search completion:(void(^)(NSMutableArray<Item*> *items, NSError *error))completion
```

Comment:
This method is almost identical to the one above. What you can do is pass the `path` as a parameter to this method and reuse above. In the first call the path won't have the cursor and in the second page you can include the cursor.